src/clr-angular/data/datagrid/render/header-renderer.ts

```diff
@@ -42,6 +59,23 @@ export class DatagridHeaderRenderer implements OnDestroy {
   private subscriptions: Subscription[] = [];
   ngOnDestroy() {
     this.subscriptions.forEach(sub => sub.unsubscribe());
+    if (this.stateSubscription) {
+      this.stateSubscription.unsubscribe();
+    }
+  }
+
+  private stateChanges(state: DatagridColumnState) {
+    if (state.actions && state.actions.length) {
+      state.actions.forEach(action => {
```

Comment:
Would it make sense to add a util method to the columns service that can centralize the case detection for DatagridColumnsChanges type instead of repeating them in all of the renderers?

Reply:
There are some duplicated lines in the renderers now that I am thinking I'll extract over time as the need arises, for now I wasn't too worried about it as it makes this change more complex.